qt_gui_cpp/CMakeLists.txt

```diff
@@ -19,14 +19,62 @@ if(WIN32)
 endif()
 
 find_package(pluginlib REQUIRED)
+find_package(Python3 REQUIRED COMPONENTS Development)
+find_package(Qt5 REQUIRED COMPONENTS Widgets)
 find_package(rcpputils REQUIRED)
 find_package(tinyxml2_vendor REQUIRED)
 find_package(TinyXML2 REQUIRED)
 
 ament_python_install_package(${PROJECT_NAME}
   PACKAGE_DIR src/${PROJECT_NAME})
 
-add_subdirectory(src)
+set(qt_gui_cpp_SRCS
+  src/qt_gui_cpp/composite_plugin_provider.cpp
+  src/qt_gui_cpp/generic_proxy.cpp
+  src/qt_gui_cpp/plugin_bridge.cpp
+  src/qt_gui_cpp/plugin_context.cpp
+  src/qt_gui_cpp/plugin_descriptor.cpp
+  src/qt_gui_cpp/plugin_provider.cpp
+  src/qt_gui_cpp/recursive_plugin_provider.cpp
+  src/qt_gui_cpp/settings.cpp
+)
+
+set(qt_gui_cpp_HDRS
+  include/qt_gui_cpp/plugin.h
+  include/qt_gui_cpp/plugin_bridge.h
+  include/qt_gui_cpp/plugin_context.h
+)
+
+qt5_wrap_cpp(qt_gui_cpp_MOCS ${qt_gui_cpp_HDRS})
+
+add_library(${PROJECT_NAME} SHARED ${qt_gui_cpp_SRCS} ${qt_gui_cpp_MOCS})
+target_include_directories(${PROJECT_NAME} PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include>")
+
+target_link_libraries(${PROJECT_NAME}
+  ${QT_QTCORE_LIBRARY}
+  ${QT_QTGUI_LIBRARY}
+  Qt5::Widgets
+  pluginlib::pluginlib
+  Python3::Module
+  rcpputils::rcpputils
+  tinyxml2::tinyxml2)
+
+if (WIN32)
+  # On Windows systems, Visual Studio does not currently set __cplusplus correctly unless this
+  # compile option is added. __cplusplus is used by a header in pluginlib/class_loader.hpp
+  message(STATUS "On Windows, setting /Zc:__cplusplus")
```

Comment:
pluginlib should be exporting the compile definition on Windows, instead of adding them in all dependencies

Reply:
Yeah that's interesting. It was added way back in https://github.com/ros-visualization/qt_gui_core/pull/146 .

It appears connected to this: https://github.com/ros/pluginlib/blob/0c28429c8702c30d630c7335ee2823e2382bf92d/pluginlib/include/pluginlib/class_loader.hpp#L47 - which looks like pluginlib conditionally includes `<memory>`. I would think it's time to make `pluginlib` require c++11 instead of conditioning code on it.

Reply:
I opened an upstream PR to remove the using of this define, and also it appears like the upstream header shouldn't require this anyways because it uses `_MSC_VER` instead of `__cplusplus` on Windows.

Reply:
Removed this check in eef1a62